### PR TITLE
[AAE-6382] User acronym is not displayed

### PR DESCRIPTION
**Pull request by BSekula**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://alfresco.atlassian.net/browse/AAE-6366

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/pipes/user-initial.pipe.spec.ts [hunk 1]
```diff
@@ -88,6 +89,17 @@ describe('UserInitialPipe', () => {
...
 
+    it('should return a single letter from username if we do not have first and last name', () => {
+        const fakeIdentityUser: IdentityUserModel = {
+            firstName: undefined,
+            lastName: undefined,
+            username: 'FAKE-USERNAME'
+        };
+
+        const result = pipe.transform(fakeIdentityUser);
+        expect(result).toBe('<div id="user-initials-image" class="">F</div>');
```
Comment (mauriziovitale) [inline lib/core/pipes/user-initial.pipe.spec.ts:100]: what if the username is lowercase ?

File: lib/core/pipes/user-initial.pipe.ts [hunk 2]
```diff
@@ -34,15 +34,25 @@ export class InitialUsernamePipe implements PipeTransform {
...
+                delimiter,
+                user.username
+            );
             safeHtml = this.sanitized.bypassSecurityTrustHtml(`<div id="user-initials-image" class="${className}">${initialResult}</div>`);
         }
         return safeHtml;
     }
 
-    getInitialUserName(firstName: string, lastName: string, delimiter: string): string {
+    getInitialUserName(firstName: string, lastName: string, delimiter: string, username?: string): string {
```
Comment (mauriziovitale) [inline lib/core/pipes/user-initial.pipe.ts:48]: I don't like this solution very much. 
Why don't we create a getInitials for unsername and one for firstnameLastname ?

Comment (mauriziovitale) [reply lib/core/pipes/user-initial.pipe.ts:48]: another point that is not clear to me is from where this displayName comes from, I know that it was already there... but is very confusing

**Review by mauriziovitale — CHANGES_REQUESTED** [review]
comments

File: lib/core/pipes/user-initial.pipe.spec.ts (hunk 1 above)
Comment (BSekula) [reply lib/core/pipes/user-initial.pipe.spec.ts:100]: Transform text to uppercase (for `firstName` and `lastName` also)

File: lib/core/pipes/user-initial.pipe.ts (hunk 2 above)
Comment (BSekula) [reply lib/core/pipes/user-initial.pipe.ts]: Refactored the code.

We can see now exactly from where we are getting these properties.

Other solution would be just to add:

```
interface User {
    firstName?: string;
    lastName?: string;
    displayName?: string;
    username?: string;
}
```

File: lib/core/pipes/user-initial.pipe.spec.ts (hunk 1 above)
Comment (mauriziovitale) [reply lib/core/pipes/user-initial.pipe.spec.ts:99]: ok my point is to have ```username: 'fake-username'```

**Pull request closed without merging**